R/download_package.R

```diff
@@ -0,0 +1,327 @@
+#' Remove and substitute special characters in a dataOne identifier
+#'
+#' This is a helper function for the 'download_package' function. Remove special
+#' characters from a package identifier for use as a filename prefix. Can be
+#' generalized for use with any pid.
+#'
+#' @author Dominic Mullen, \email{dmullen17@@gmail.com}
+#'
+#' @param pid (character) The identifier a dataOne object
+#'
+#' @return (character) The formatted identifer as a string
+pid_to_prefix <- function(pid) {
+    pid <- gsub(":", ";", pid)
+    pid <- gsub("\\/", "_", pid)
+    pid <- gsub("\\.", "$", pid)
+
+    return(pid)
+}
+
+#' Convert excel workbook to multiple csv files
+#'
+#' This is a helper function for download_package.
+#'
+#' @param path (character) File location of the excel workbook.
+#' @param prefix (character) Optional prefix to prepend to the file name.
+#'
+#' @author Dominic Mullen \email{dmullen17@@gmail.com}
+#'
+#' @return (invisible())
+excel_to_csv <- function(path,
+                         prefix = NULL) {
+    stopifnot(file.exists(path))
+    #stopifnot(is.character(prefix))
+
+    # Stop if the user doesn't have the readxl package installed
+    if (!requireNamespace("readxl")) {
+        stop(call. = FALSE,
+             "The readxl package is required to convert excel workbooks to csv. Please install it and try again.")
+    }
+
+    # Try to read excel file and split into csvs
+    tryCatch({
+        sheets <- excel_sheets(path)
+
+        lapply(seq_along(sheets), function(i) {
+            csv = read_excel(path, sheet = sheets[i])
+
+            if (!is.null(prefix)) {
+                file_path <- file.path(dirname(path), paste0(prefix, sheets[i], ".csv"))
+            } else {
+                file_path <- file.path(dirname(path), paste0(sheets[i], ".csv"))
+            }
```

Comment:
You may want to may want to run your pid_to_prefix function on sheet names, in case they contain special characters.  (Just tried it out in excel, and it let me add ";" to the sheet names.)   If you do, you could rename it to remove_special_characters or something more general

Reply:
i like the new name!